File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/DateIntervalCreateFromDateString.java
package com.onevgo.functions;

import java.time.Duration;

public class DateIntervalCreateFromDateString {
    public static void main(String[] args) {
        System.out.println(Duration.parse("P1D"));
//        System.out.println(Duration.parse("P2W"));
//        System.out.println(Duration.parse("P3M"));
//        System.out.println(Duration.parse("P4Y"));
        System.out.println(Duration.parse("P1DT12H"));
        System.out.println(Duration.parse("PT3600S"));
    }
}

File: 01_Language/01_Functions/php/date_interval_create_from_date_string.php
<?php

// Each set of intervals is equal.
$i = new DateInterval('P1D');
$i = DateInterval::createFromDateString('1 day');

$i = new DateInterval('P2W');
$i = DateInterval::createFromDateString('2 weeks');

$i = new DateInterval('P3M');
$i = DateInterval::createFromDateString('3 months');

$i = new DateInterval('P4Y');
$i = DateInterval::createFromDateString('4 years');

$i = new DateInterval('P1Y1D');
$i = DateInterval::createFromDateString('1 year + 1 day');

$i = new DateInterval('P1DT12H');
$i = DateInterval::createFromDateString('1 day + 12 hours');

$i = new DateInterval('PT3600S');
$i = DateInterval::createFromDateString('3600 seconds');
